cmd/web-handlers.go

```diff
@@ -683,30 +686,36 @@ next:
 				}) {
 					return toJSONError(ctx, errAccessDenied)
 				}
-				if globalIAMSys.IsAllowed(iampolicy.Args{
-					AccountName:     claims.AccessKey,
-					Action:          iampolicy.BypassGovernanceRetentionAction,
-					BucketName:      args.BucketName,
-					ConditionValues: getConditionValues(r, "", claims.AccessKey, claims.Map()),
-					IsOwner:         owner,
-					ObjectName:      objectName,
-					Claims:          claims.Map(),
-				}) {
-					govBypassPerms = ErrNone
-				}
-				if globalIAMSys.IsAllowed(iampolicy.Args{
-					AccountName:     claims.AccessKey,
-					Action:          iampolicy.GetBucketObjectLockConfigurationAction,
-					BucketName:      args.BucketName,
-					ConditionValues: getConditionValues(r, "", claims.AccessKey, claims.Map()),
-					IsOwner:         owner,
-					ObjectName:      objectName,
-					Claims:          claims.Map(),
-				}) {
-					govBypassPerms = ErrNone
+
+				// Check if user is allowed to run BypassGovernanceRetention
+				// and GetBucketObjectLockConfiguration actions
+				if bucketObjectLockEnabled {
+					if !globalIAMSys.IsAllowed(iampolicy.Args{
+						AccountName:     claims.AccessKey,
+						Action:          iampolicy.BypassGovernanceRetentionAction,
+						BucketName:      args.BucketName,
+						ConditionValues: getConditionValues(r, "", claims.AccessKey, claims.Map()),
+						IsOwner:         owner,
+						ObjectName:      objectName,
+						Claims:          claims.Map(),
+					}) {
+						govBypassPerms = ErrAccessDenied
+					}
+					if globalIAMSys.IsAllowed(iampolicy.Args{
+						AccountName:     claims.AccessKey,
+						Action:          iampolicy.GetBucketObjectLockConfigurationAction,
+						BucketName:      args.BucketName,
+						ConditionValues: getConditionValues(r, "", claims.AccessKey, claims.Map()),
+						IsOwner:         owner,
+						ObjectName:      objectName,
+						Claims:          claims.Map(),
+					}) {
+						govBypassPerms = ErrAccessDenied
```

Comment:
IsAllowed doesn't mean its Deny! @ebozduman the logic is not correct here.

Reply:
I am not sure how I made you get that idea, but I do not think `IsAllowed` means `Deny`. 
Do you mean `IsAllowed` logic is wrong or my fix of putting them inside another check which makes sure there is a retention/lock setting for the bucket is created and active.
That is because there is no point checking those 2 actions if there is no active retention/lock configuration. So, why bother doing that check when there is nothing configured on the bucket/object.

Reply:
no, what you wrote is wrong you are using IsAllowed to set Deny i.e ErrAccessDenied - which looks incorrect because your intention is to allow not deny.